pilot/pkg/proxy/envoy/v1/config.go

```diff
@@ -369,7 +369,21 @@ func buildRDSRoute(mesh *meshconfig.MeshConfig, node model.Node, routeName strin
 			targetGateways[spec.Name] = true
 		}
 
+		finalConfig := &HTTPRouteConfig{}
 		httpConfigs = buildGatewayHTTPRoutes(mesh, node, instances, services, config, targetGateways)
+		for _, config := range httpConfigs {
+			for _, host := range config.VirtualHosts {
+				vhost := &VirtualHost{
+					Name:    host.Name,
+					Routes:  host.Routes,
+					Domains: host.Domains,
+				}
+
+				finalConfig.VirtualHosts = append(finalConfig.VirtualHosts, vhost)
```

Comment:
This is the most import bit that now allows us to decouple the envoy port from the backing service port.  The special bit here is we don't end up stripping out a host with no port. So, we end up with domains of `pivotal.io:ServicePort` and `pivotal.io`. The `pivotal.io` does end up routing the traffic to the right cluster but as @kyessenov explained this only works if all of your backing services have the same port (fine for now). We'd need to solve this for multi-port services but this may be fine at present.

This also breaks the weighted-routing case but I haven't quite figured out why.